docs/guide/using-ros.md

```diff
@@ -89,8 +89,8 @@ In this case, you can build your own custom ROS controller.
 
 It is possible to implement such a ROS node in C++ using the "roscpp" library on Linux and macOS.
 However, in this case, you need to setup a build configuration to handle both the "catkin\_make" from ROS and the "Makefile" from Webots to have the resulting binary linked both against the Webots "libController" and the "roscpp" library.
+An example is provided [here](https://github.com/cyberbotics/webots/tree/master/projects/vehicles/controllers/ros_automobile) to create a specific controller for controlling a vehicle.
 
-On Windows, Linux and macOS, such a controller can be implemented in Python by importing both ROS libraries (including [rospy](http://wiki.ros.org/rospy)) and Webots libraries (controller) in a Webots robot or supervisor controller.
-A simple example of a Python custom ROS controller is provided in the [`WEBOTS_HOME/projects/languages/ros/controllers/ros_python/`](https://github.com/cyberbotics/webots/tree/master/projects/languages/ros/controllers/ros_python) folder.
-A `README.md` file provides all the necessary details to understand it and further extend it.
-A [second more complicated example](https://github.com/cyberbotics/webots/tree/master/projects/robots/universal_robots/controllers/universal_robots_ros) shows how to interface a model of a Universal Robots arm in Webots with ROS using [rospy](http://wiki.ros.org/rospy).
+An even more generic solution, is to use an [extern controller](running-extern-robot-controllers.md) and run the controller as a regular ROS node on the ROS side.
+A very simple example is provided [here](https://github.com/cyberbotics/webots/blob/master/projects/languages/ros/webots_ros/scripts/ros_python.py), it is written in pure Python and should work on Windows, Linux and macOS, straight out of the box.
```

Comment:
Maybe the corresponding launch file could be mentioned and explained (`ros_controller.py` does "obstacle avoidance"). Users may find it more useful if they can observe the behaviour first and see the full project (that includes launch file).